core/scan_result.cxx

```diff
@@ -32,17 +32,19 @@ class scan_result_impl
 
     [[nodiscard]] auto next() const -> tl::expected<range_scan_item, std::error_code>
     {
-        if (auto item = iterator_->next().get(); item) {
-            return item.value();
-        }
-        return tl::unexpected{ errc::key_value::range_scan_completed };
+        return iterator_->next().get();
     }
 
     void next(utils::movable_function<void(range_scan_item, std::error_code)> callback) const
     {
         return iterator_->next(std::move(callback));
     }
```

Comment:
I'm somewhat concerned with what would happen in the case where a stream gets retried indefinitely. If the server returns an EBusy status, the scan is set to be retried as per the RFC, but there's no limit for how long it should be retried for. As timeouts only apply to the range_scan_continue and range_scan_create operations, does this mean that calling next() could theoretically lead to a very long or indefinite wait (for example if the server for whatever reason keeps returning EBusy statuses)?

Reply:
RFC has this in the "Error Handling & Retry Strategy"

> For non-sampling scans, all effort should be made to keep the range scan streams going until timeout or completion. This also includes restarting individual partition sub-streams if at all possible (based on the errors outlined below).

I guess it could be interpreted that the global timeout will force completion of all the streams with timeout status or something like that.

Reply:
The check for the timeout before starting a stream should now resolve this